com.unity.netcode.gameobjects/Runtime/Messaging/CustomMessageManager.cs

```diff
@@ -52,13 +52,18 @@ internal void InvokeUnnamedMessage(ulong clientId, FastBufferReader reader)
         /// <param name="clientIds">The clients to send to, sends to everyone if null</param>
         /// <param name="messageBuffer">The message stream containing the data</param>
         /// <param name="networkDelivery">The delivery type (QoS) to send data with</param>
-        public void SendUnnamedMessage(List<ulong> clientIds, FastBufferWriter messageBuffer, NetworkDelivery networkDelivery = NetworkDelivery.ReliableSequenced)
+        public void SendUnnamedMessage(IReadOnlyList<ulong> clientIds, FastBufferWriter messageBuffer, NetworkDelivery networkDelivery = NetworkDelivery.ReliableSequenced)
```

Comment:
this and the other method below have API breaking changes in this PR, I wonder if that's intentional and worth it here (hence I changed the title to `fix!:` from `fix:`).

if we're going to make an API breaking change, (also to @TheCaveOfWonders's point), we might just have a separate method (method overload) that takes `clientIds` and another one that doesn't. also, putting `clientIds` parameter to the end similar to `networkDelivery` would allow us to make it an optional parameter as well.

I think we could think a little bit more about API design here if we really want to go for API breaking changes. Last thing to mention is that, I believe we should also xmldoc things properly so that it would also become clear to users what the behavior will be if they pass an empty vs null `clientIds` to these methods.

_(and, is this critical/needed/targeting 1.0 release?)_

Reply:
Technically you're correct that this is a breaking change... but the ticket that was assigned to me here was "documentation says it does this, but it doesn't," so I would argue that it's not a breaking change, it's a fixing change, because the documentation has long said that null = everything here, and it used to work this way.

The ticket that was assigned to me was from hackweek, so my assumption was that it is for the 1.0 release.

Reply:
Although if you're referring to the change from List to IReadOnlyList, that's not a breaking change, because List implements IReadOnlyList, so it's still compatible with existing implementations.